### DISCUSSION TITLE
Sequent RS485 wants to work but isn't quite there?

### DISCUSSION THREAD

**Discussion by StrikerS7R** [tagyoureit/nodejs-poolController#581] [RS485] [answered]
Been super happy with the full Nixie standalone setup the past few months and actually relocated and rebuilt my whole pad including a new iC40 salt cell and heat pump heater. I have it set up and the pump/temp side works great but I'm having issues getting the salt cell to communicate.

I've been digging through all the threads I can find around it the past week and have to admit defeat and just ask for help from the true pro's. Lots of useful items I found though from those threads and seems like you all managed to get those folks up but those fixes aren't working for my setup for some reason.

![image](https://user-images.githubusercontent.com/22134823/187265663-b0620ded-2f44-4e77-a180-18f1de5553ba.png)

I'm using the recommended Subsequent 8 relay and building automation hats for my rPi4 and have wired the rs485 cables from the first port on the automation card and have the 2 jumpers installed. 

**_**One maybe unique item that might be a cause if I've missed a config item is that I have the relay hat first and the automation card on top (so I can do the other input jumpers as the system grows). Accordingly my board level is 1 (not the default 0 in the wiki).**_** Both cards seem to work fine otherwise though so....?

I've installed the latest MEGABAS updates (v1.2.2 and 1.09 FW) with it set to 0 38400 1 0 2 (I think the last address item is right per the [Subsequent git](https://github.com/SequentMicrosystems/megabas-rpi/blob/master/MODBUS.md) slave address offset) and have it set to passthrough in REM with the port set to /dev/ttyS0.

I do get the log file looking alright with Serial port: /dev/ttyS0 request to open successful 9600b 8-none-1  

Then it gets stuck in an ongoing loop with the string below and id value incrementing up by 1 every 4 for the first packet and every 8 for the second
info: {"port":0,"id":9,"valid":true,"dir":"out","proto":"chlorinator","pkt":[[],[], [16,2,80,0], [0],[98,16,3]],"ts":"2022-08-29T10:47:02.754-0700"}
info: {"port":0,"id":104,"valid":true,"dir":"out","proto":"chlorinator","pkt":[[],[], [16,2,80,17], [50],[165,16,3]],"ts":"2022-08-29T10:59:32.520-0700"}
![image](https://user-images.githubusercontent.com/22134823/187268304-ebf75e71-b1fe-4367-9f7a-aab1d32211df.png)

- I've tried swapping the A/B wires but no change detected. Was careful to power down both systems and not touch wires when swapping
- I actually tried using the other two wires to send +15VDC (red) and 0VDC (black) to the terminals on the IntelliChlor power center like as if it were a Pentair controller but also no change (left them off for now). I do have the green light on the power center card as well. 
- Tried rebuilding the whole system actually deleting the full directories for all 3 and re-pulling fresh but no change either. 
- Tried changing both the megabas baud to 9600 to match the njsPC settings and the njsPC side to 38400 per megabas but also no change other than the obvious recalculated stats (nice touch btw!)
- Moved the green/yellow comms wires from the terminal blocks directly to the card jack as well to be sure of good connections. 
- I don't have any other rs485 devices to test with yet to fully confirm it's not the iChlor card and am waiting on electrician to pull feeds to the heater but unknown timeline on that. My pump is a SuperFlo VS so I have 4 of the relays controlling the speeds (no rs485 comms there)
- The three package installs were done with node v12.22.0 as they seemed the most happy with that (lots of engine errors in the install when using the default v12.18.3 or the more recent v16.17.0 or v18.8.0

Unsure where else to look or test for a path to resolve, would love some assistance from other more knowledgeable as this has been my most involved raspberry pi project by a long shot. I'm quite handy mechanically and electrically and have been working on my programming skills but there's a long way to go on that front.

Thanks in advance!!

**Comment by StrikerS7R** [comment]
Also didn't note above but I did confirm from the other threads that have the Pi config set to:
SPI: Enabled
I2C: Enabled
Serial Port: Enabled
Serial Console: _Disabled_
1-Wire: Enabled
Remote GPIO: Disabled

**Reply by rstrouse** [reply]
Lets start with some easy peasy stuff.  So the board level for the BAS/Relay card are of no significance.  The board level is just a fancy way for Sequent to obfuscate the I2c address.  The starting address between these two boards are way away from each other so there is no need to set the jumper unless you have two of either board.

From the display above, the comms do indicate a reversal of the wires.  This typically manifests itself as bytes received but no successful inbound messages and 100% failure for any outbound messages.  The repeated loop you are seeing in the messageManager is njsPC trying to contact IntelliChlor but not receiving any discernable response.

From a pi console window please run the following commands.
`cd ~/megabas-rpi`
`megabas 1 board`

This should verify the board is talking.

Next shoot me a screenshot of your I2c tab.

**Reply by rstrouse** [reply]
Also run the following command from a Raspberry Pi prompt and post the results here.

`ls -l /dev/ttyS0`

**Reply by StrikerS7R** [reply]
Sure thing, that's good to hear on the hat front. I can try to swap the wires again as it was done as a step amongst many others in try to self troubleshoot so may have done while I broke something else but I do know that I currently have the YLW/GRN terminals on the cell matched to the same wire colors and terminated directly yellow to A and green to B.

![image](https://user-images.githubusercontent.com/22134823/187287869-bca7f023-a1cf-438e-8cee-6b8d5aae3615.png)

![image](https://user-images.githubusercontent.com/22134823/187288365-03a0acec-7012-4291-a825-cf653f9b5bf3.png)
![image](https://user-images.githubusercontent.com/22134823/187289286-a81a4e34-03a7-448d-bf05-b15903c8b367.png)

And a slight tangent from following the posted guide, I do **not** get the desired ffa5ff 
![image](https://user-images.githubusercontent.com/22134823/187288726-09489165-cedd-4155-bfa9-b9aec2e05f82.png)

Thanks again for the attention! Not just fluffing for the help but I really do appreciate the effort you've put into this entire package and must say it's impressive.

**Reply by rstrouse** [reply]
Thanks for the compliments.  Swap the wires on the connector that is plugged into the BAS.   Is this a Raspberry Pi 3 or 4?  Try turning off Bluetooth.  The BT implementation on some pi versions actually hijacks the serial uart.

**Reply by rstrouse** [reply]
BTW you will not see the ffa5ff in your output.  That only appears when there is a Pentair OCP or RS485 pump/heater attached.

**Reply by tagyoureit** [reply]
> Been super happy with the full Nixie standalone setup the past few months and actually relocated and rebuilt my whole pad including a new iC40 salt cell and heat pump heater.

Can you also include exact details of how you are wiring your mega-bas up to your chlorinator?  Pics would be helpful.

**Reply by StrikerS7R** [reply]
No problem. It's a Pi 4 and I turned off Bluetooth (it was on) and swapped the rs485 connector wires on the BAS end. Repowered both the Pi and the salt cell power center and let them come online.

Unfortunately doesn't look like anything changed that I can tell. I get the same /dev/ttyS0 request to open successful 9600b 8-none-1 then the looping messages in the message manager and port stats.
![image](https://user-images.githubusercontent.com/22134823/187320536-e26a8c80-dd8d-443a-8b28-84a1b3034b6e.png)

No problem, here it is now with the wires to the salt cell power center directly terminated on the BAS card
![image](https://user-images.githubusercontent.com/22134823/187320497-902cad81-16bb-4bf5-9044-89cd65ddd0eb.png)
![image](https://user-images.githubusercontent.com/22134823/187320517-644c3c51-2f01-4aba-9f58-df0c7bdddb06.png)

**Reply by StrikerS7R** [reply]
Also to clarify that power cell side photo slightly. The red and brown (in lieu of black) wires are not terminated on the Pi end. I had those briefly when trying to send the +15VDC & 0 VDC that its manual indicated wondering if that would power the comms card. Just spit balling at that point but it's been disconnected for a while

**Comment by rstrouse** [comment]
Run the following commands
`cd ~/megabas-rpi`
`megabas 1 rs485wr 0 9600 1 0 1`
`megabas 1 rs485rd`

On the last command make sure it echos back the settings.

Also keep in mind the cell will not respond until it has booted completely.  This can take a full minute.  Also what are the red/brown wires connected to?  These should not be needed in this case.

**Reply by StrikerS7R** [reply]
Done

![image](https://user-images.githubusercontent.com/22134823/187336834-9d99a469-4e13-4ff3-ac09-7ebd926d5106.png)

Shouldn't the last bit be a 2 in my case? Since I have the BAS hat on top of the relay hat in stack level 1 with a single jumper on ID0.
From the [Subsequent git](https://github.com/SequentMicrosystems/megabas-rpi/blob/master/MODBUS.md)
     ![image](https://user-images.githubusercontent.com/22134823/187336912-9fd21927-8f4e-468a-961a-0795263dbf0a.png)

I left it as a 1, rebooted and refreshed everything but no change.
![image](https://user-images.githubusercontent.com/22134823/187337142-3222b691-c873-4a81-9d93-43f81a586c8c.png)

Just to be pre-emptive I tried again with it as a 2 but same outcome so put back to recommended 1 address

I still have the yellow/green wires reversed per last instructions. The red and brown (in lieu of black) wires are not terminated on the Pi end. I had those briefly when trying to send the +15VDC & 0 VDC that its manual indicated wondering if that would power the comms card. Just spit balling at that point but it's been disconnected for a while. Here's the whole box in it's untidy until complete state for reference. 
![image](https://user-images.githubusercontent.com/22134823/187337477-a7f1027d-53f6-4e53-908c-530218e4a8c8.png)

**Reply by rstrouse** [reply]
Hmmm run npm i in the nodejs-poolController directory.  Watch for binary errors related to the serial port.  We have a real mystery on our hands here.

**Reply by rstrouse** [reply]
The slave address they are referring to there is a modbus slave we are disabling modbus on the card.  I wonder though if you should set it to stack level 0.  The position of the card on the connector is irrelevant.  The stack level that they are talking about with the jumpers is simply the I2c address.  Remove the stack jumper and delete the card from REM then re add it at the new address.

**Reply by StrikerS7R** [reply]
Ah interesting, learning new things with this project for sure.

First, replying to your last about running npm i and looking for errors. I did get several errors but all related to eslint dependencies
![Screenshot_20220829-213921_VNC Viewer](https://user-images.githubusercontent.com/22134823/187351012-ea6badae-0894-4586-83ef-4272b9ae6c68.jpg)

Second, on removing the jumper and card from REM and re-adding it back at the new address, I presume I should also reconfigure the megabas rs485 at the 0 stack level too right? (Via "megabas 0 rs485wr 0 9600 1 0 1")

**Reply by rstrouse** [reply]
Could I get you to run `node -v`  what version does it report.

**Reply by StrikerS7R** [reply]
Jumper removed, BAS deleted and re-added to REM. Confirmed it was set to RS485 pass-through and reconfigured one of the temp sensor feeds to verify the card was still working. Those worked fine but still same story on the RS485 comms, 0 received and 100% failure on send. I confirmed the megabas settings and they stayed the same and followed the board to id 0 so no change needed there.
![image](https://user-images.githubusercontent.com/22134823/187486018-63beabe9-43a3-4c28-bfae-167035dee290.png)

It reports node version 12.18.3.
![image](https://user-images.githubusercontent.com/22134823/187486885-aeb66290-1147-4e82-8b0b-090e94d6249a.png)

Full disclosure, I did change the node version during the install as well as update npm similar to [this thread on TFP](https://www.troublefreepool.com/threads/nodejs-pool-controller-work-with-pump-only.246189/post-2164041) when you were helping john800xc, but it looks like it's reverting back to node v12.18.3 after a reboot.

Nunning a NPM audit within nodejs-poolController does give a high severity crash in HeaderParser in dicer but not sure that's related
![image](https://user-images.githubusercontent.com/22134823/187488126-e03d08c5-f02b-43b9-92ce-5b77e784b3db.png)

At this point if you think it worthwhile I'm game to wipe the whole thing and set it up fresh since I'm way more familiar now vs. my first bumbling through and may have misconfigured something along the way

**Reply by rstrouse** [reply]
I think we may have serial port binaries installed for a later version of node but are running an older version of node.

Try this

`nvm use 14.20.0`
`cd ~/nodejs-poolController`
`npm rebuild`

**Reply by StrikerS7R** [reply]
That's definitely promising, I got a whole bunch of serial port errors during that rebuild
![image](https://user-images.githubusercontent.com/22134823/187496021-f541d9b5-3370-4284-8407-eb11da6fe9a1.png)
![image](https://user-images.githubusercontent.com/22134823/187496111-fe2aca77-0be1-40a4-a5e5-aab52e5104ee.png)

**Reply by rstrouse** [reply]
Ok run it like this.

`cd ~/nodejs-poolController`
`git pull`
`npm i`
`npm rebuild`

**Reply by StrikerS7R** [reply]
that went much better
![image](https://user-images.githubusercontent.com/22134823/187503377-daac2127-ed44-42e9-b416-7423dbd03851.png)

**Reply by StrikerS7R** [reply]
So the above steps didn't change the result of the rs485 comms only active on the Pi side. 

I decided to give it a fresh start and used a spare SD to format a fresh Pi install and start from scratch. It did go much smoother as I started with the current 0.39.1 nvm and only installed npm 6.14.17 with node v14.20.0

However.... when doing the initial `npm i` on njsPC I got the exact same series of warnings on @serialport/bindings that there are no pre-built binaries found on target node v14.20.0. 

> That's definitely promising, I got a whole bunch of serial port errors during that rebuild ![image](https://user-images.githubusercontent.com/22134823/187496021-f541d9b5-3370-4284-8407-eb11da6fe9a1.png) ![image](https://user-images.githubusercontent.com/22134823/187496111-fe2aca77-0be1-40a4-a5e5-aab52e5104ee.png)

I did the `npm rebuild` and `npm i` again and those warnings didn't come up so I assume those warnings was it building the binaries that it had previously warned weren't already pre-built?

Anywho, it seemed fine so I left alone and carried on following the wiki instructions and got it all set up back to how the old card had it with the pump, temp sensors, relays all set up and configured ...and with apprehension... the chlorinator. 

I feel like I'm building for a happy ending but alas it behaved the exact same way as before. 100% failure rate
![image](https://user-images.githubusercontent.com/22134823/188019037-6b21dd8d-385e-4778-91d0-883be1b037ec.png)

With a tiny bit of hope I powered everything down and swapped the green/yellow wires at the BAS connector and rebooted it all back up. Still no dice, same 100% failure rate. I even tried waiting until the Nixie setup was all online before powering up the chlorinator power center in case it only looked for active comms for a brief period after startup as supposed in another thread but still no change.

Any other ideas to try on possible sources or solutions? At this point I have a backup SD card already configured so if I hose up one trying something odd I'm not offline for long.

**Reply by tagyoureit** [reply]
Serialport rebuilding the binaries is expected. 

Here are three things you can try:
1. Try the other rs485 port on the BAS card
2. Check continuity or try a different set of rs485 wires to make sure there is no break in the physical wires
3. But a 2nd rs485 comms device. This would let you try a few things including bypassing the BAS card and snooping to see if traffic is actually being sent out like njsPC says. 
4. And a bonus, try to use any other rs485 device that you own and see if you have success. The chlorinators can be the most finicky to get working.

**Reply by StrikerS7R** [reply]
New [USB RS485 card](https://smile.amazon.com/gp/product/B081MB6PN2/) inbound to try that route, other port and alternative wire run didn't change anything. 

I don't have any other rs485 devices (unless the Subsequent relay hat counts but not sure how that would work with it already stacked on the GPIO pins).

> Serialport rebuilding the binaries is expected.
> 
> Here are three things you can try:
> 
> ~~1. Try the other rs485 port on the BAS card~~
> ~~2. Check continuity or try a different set of rs485 wires to make sure there is no break in the physical wires~~
> 3. But a 2nd rs485 comms device. This would let you try a few things including bypassing the BAS card and snooping to see if traffic is actually being sent out like njsPC says.
> 4. And a bonus, try to use any other rs485 device that you own and see if you have success. The chlorinators can be the most finicky to get working.

**Reply by rstrouse** [reply]
Wait a minute.  Do you have a jumper installed at the leftmost pin (R) on J2 of the 8-relay card?  If you do, then there is the problem.  Only one ttyS0 can be used for the main UART of the Raspberry pi.

**Reply by StrikerS7R** [reply]
Ooo, interesting. I don't think I do but will check in the morning. To be explicit, I should not have that jumper installed on the relay hat (nor on the BAS hat per your earlier comment about the addressing being so different anyway) right? 

Actually, this brings up an earlier "mistake" (is it?) I made by plugging the yellow/green rs485 wires from the chlorinator into the rs485 port on the 8-relay card before later moving it to the BAS card ports after researching the issue. Definitely a knowledge gap area for me (obviously) so not sure how these pieces work together especially with the handshake comms.

I do have another ttyAMA0 show up, but no idea if that's helpful to diagnosing this, just an FYI if it means something to you

**Reply by rstrouse** [reply]
Double check to make sure you do not have a jumper installed on the j2 for the 8 relay.  If you have more than one 8 relay then you choose the address by adding a jumper on the next jumper but not the one labeled R.  You do not need to add an address jumper to either the MEGA-BAS or the relay 8 if you have only one of each.  You only need jumpers on the rs485 pins for the MEGA-BAS.

**Reply by StrikerS7R** [reply]
Confirmed no jumper on the 8-relay hat (only 1 relay hat and 1 BAS hat) and only 2 jumpers on the BAS on the J3 pins. 1 across the TX pins & 1 across RX pins.

I pulled off and reseated those two jumpers but no change to the comms. The rs485 USB dongle arrives this evening so I'll give that a go and see what it does.

**Reply by rstrouse** [reply]
Let me know how it goes.  It could be that you simply have a defective RS485 port on the BAS.

**Reply by StrikerS7R** [reply]
So I think we may have identified the culprit there.

Plugged in the new RS485 dongle, changed the njsPC config to reference ttyUSB0, connected the wires up (then swapped them after only seeing the TX light), started njsPC back up and it happily started talking to the chlorinator. 

![image](https://user-images.githubusercontent.com/22134823/189401206-cd592aae-3ee6-451c-9201-e547b71776b3.png)

Seems odd that the RS485 on the BAS is bad out of the box (only on RX?) but guess it's possible. One other question I had for you based on your comment to make sure there's not a J2 jumper on the relay hat. Does that mean that now I know the BAS one is bad, I could use the RS485 on the relay hat instead of the dongle? Again showing my ignorance but I'm fine using the dongle if that's the best solution.

**Reply by rstrouse** [reply]
Yes if you remove the RS485 jumpers from the BAS and install the J2 (R) jumper on the Relay 8 you can use the RS485 on the Relay 8 card.  It should hook the serial UART for RS485 through the Relay 8.  RS485 is differential voltage comms so if either wire is not pulled high or low you will get a situation where the buffer output appears to have sent a request but it never really translates to anything but garbage on the wire.  Data will go out but it only be a half truth.

**Reply by tagyoureit** [reply]
In your pic above you have 1,322 reconnects.  Did you install the new adapter and switch to it without restarting njsPC?  That should be close to 0 with a chatty bus.  The send aborted/retries are also quite high.  If you didn't restart then no worries.  If you did restart and still seeing those high errors I would double click into that still.

**Reply by StrikerS7R** [reply]
I had stopped njsPC but not REM or the dashboard (via pm2 stop) during the config file change. It doesn't appear that either of those stats have change at all today though (exact same Reconnects, Aborted/Retries as this morning). I would assume that implies the current connection is stable with the USB dongle.

![image](https://user-images.githubusercontent.com/22134823/189457674-56c49e6f-1461-436d-aa90-bf60043e74b7.png)

**Reply by rstrouse** [reply]
That is correct.  The likelihood is that you had the adapter disconnected at some point and it simply did not have any traffic so it simply reconnected every 10 seconds.  If you save the port it will reset these stats.  The good numbers are there are 56k successful sends and what looks like 56k successful replies.

**Comment by StrikerS7R** [comment, marked as answer]
> Let me know how it goes. It could be that you simply have a defective RS485 port on the BAS.

Looks that way, not sure how to fully test the BAS RS485 using my new USB RS485 dongle but when directing njsPC to use the new USB one it works great.

Thanks!

**Reply by Telemachus1977** [reply]
Why are there two RS-485 ports on the BAS?  I saw the recommendation to try the other, did that make a difference?

I am by no means an expert here but you said you initially had the wires connected to the RS-485 port of the relay board.  Did you have the jumper installed to turn on RS-485 on the relay board at that time?  Could it be that it is still trying to send data to that port and not the mega-BAS (is that even possible?)

I would assume you tried the BAS without any other hat on it?

The relay board RS-485 works great.  I'm surprised it's not utilized more often.  I don't quite get the hype over the MEGA-BAS, personally.  ;)

**Reply by rstrouse** [reply]
It is actually a single port but two connectors on the BAS.  This allows more connectors if you have a lot of devices.  There are some advantages to the BAS but they must be something you are interested in.
1. There are up to 8 inputs that are pre-wired for dry contact flow switches, 10k temperature thermistors, or pressure sensors.  Any port can be configured for any function.  No additional wiring is required for this (this is probably the most important feature)
2. If you have a need for 24vac or 24vdc peripherals you can power the pi with the card using the power supply.  This would be true if you have a 24v peristaltic pump or need to control valve actuators.
3. Like the relay board it has a built in RS-485.

For the most part if you don't have a need for any of these then the relay is a great choice.  Both boards do connect to the same serial interface pins on the pi so you should only have the jumpers installed on one or the other board.  If you don't it will try to share the port and lead to a port that is talking with two mouths and listening with one ear.

The jumpers on the board are directly wired to pins 8 (GPIO14) and 10 (GPIO15) that are connected to the hardware UART on the pi.  When you remove the jumpers it disconnects the RS485 chip from these pins.

**Reply by Telemachus1977** [reply]
Maybe with undocumented changes on recent boards (dropped DC input, only 1 RS-485 port) they'll decide to give us relay output.   That would make that card an all in one solution.

**Reply by rstrouse** [reply]
Yeah if they were to take that 8-relay card and make 4 universal inputs on the front side of the card.  Or even take the BAS and remove the outputs and watchdog RTS in lieu of a few relays woohoo!  For the first time in a couple of years though it looks like everything is in stock though so maybe they will get out there to update their docs.

**Reply by Telemachus1977** [reply]
Almost you converted me over to the BAS side...

 I have a buddy who wants me to help him setup REM chem along side his easy touch.

Since he'd have 24V DC in the box for the pump, and is made of money, I figured the BAS would create a lot less work for me helping him setup his flow cell inputs and powering the pi.

But I'm not sure how to guarantee he gets the older version so I'm not going to recommend it.

**Reply by rstrouse** [reply]
I would send them an email to ask if you can still use 24vdc to power the card.  I’ll bet you can since the pi needs dc current anyway and the way to switch from ac to dc is using a bridge rectifier.  Although I do not have one I suspect they moved the smaller capacitor from the bottom of the board to create greater clearance and improve ripple reduction and that rectifier must still exist somewhere.  You can put ac or dc into a bridge rectifier and you will always get dc out.

As far as the software is concerned it doesn’t look like there is any different firmware for it so it will operate the same.

**Reply by Telemachus1977** [reply]
It might be that this would be a good use of the 4 input 4 relay board.  He'd get RS-485, the relay for his acid pump, and input for the flow switch all in one package.  

To use the optically isolated input, 24v or 5v DC could be sent to one wire of the flow switch with the other wire going to the 3v-24V input and common connected to ground without  even needing any resistors or fuses, correct?

**Reply by rstrouse** [reply]
That is correct

**Reply by Telemachus1977** [reply]
I ran njs pool controller for over a year on a raspberry Pi Zero w.   Compiling typescripts on it was excruciating, but once running it actually worked pretty well as a substitute for screen logic.

 It wasn't until I accidentally killed it while setting up REM Chem (by putting 24 volts to it instead of the intended 5 volts) that I pulled out my raspberry Pi 3B+.

I have several Zero w sitting in my parts bin.  Do you think it could run REM chem controller?

**Reply by rstrouse** [reply]
I have been threatening to get one just to try it out but I can’t think of a reason why it wouldn’t..

**Reply by Telemachus1977** [reply]
Well I have one with header pins soldered on... And a spare 4 input 4 relay board.

Bargain build REM here I come (I'll report back in an appropriate new thread)

**Reply by tagyoureit** [reply]
@Telemachus1977 Did you know you don't need to compile typescript every time?  You only need to compile it the first time you pull new code.  If you use PM2 or just follow the standard directions we use the startup script `npm start` from `package.json` which calls `npm run uild && node dist/app.js`.  There is a startup script you can use called `npm run start:cached` which skips the typescript compile step.  In the backend it runs `node dist/app.js`.  This should significantly shorten your startup time especially if you have dashPanel and REM on the same Pi.  And as a bonus, you can use the same `npm run start:cached` on all three apps.

**Reply by Telemachus1977** [reply]
Thanks.  That will be a lifesaver using and original pi zero W and 32bit OS.

Currently I can't get node js 12.8.3 (or any version of 12) to install using NVM.

pi@raspberrypi:~ $ nvm install 12.22.0
Downloading and installing node v12.22.0...
Downloading https://nodejs.org/dist/v12.22.0/node-v12.22

curl: (22) The requested URL returned error: 404

Any version up to 11 installs just fine

**Reply by StrikerS7R** [reply]
> Why are there two RS-485 ports on the BAS? I saw the recommendation to try the other, did that make a difference?
> 
Definitely agree with @rstrouse comments that the two RS485 ports on the BAS are the same as I followed the board traces to sanity check for shorts or anything unusual and they're routed the same which makes sense since it's bus style comms, just adds more connection points for additional devices. No difference was found in either port usage

>I am by no means an expert here but you said you initially had the wires connected to the RS-485 port of the relay board. Did you have the jumper installed to turn on RS-485 on the relay board at that time? Could it be that it is still trying to send data to that port and not the mega-BAS (is that even possible?)
> 
I have not had the J2 jumper installed ever and my RS485 connection to the relay board port was my oversight during assembly as I saw it marked RS485 and thought that was the referenced port in the directions but overlooked the detail about it being on the BAS until later.

> I would assume you tried the BAS without any other hat on it?
> 
I did not actually since I definitely need the relays for pump speed control (SuperFlo VS pump) 

> The relay board RS-485 works great. I'm surprised it's not utilized more often. I don't quite get the hype over the MEGA-BAS, personally. ;)

I'll probably be trying that soon as well as the dongle isn't as clean in the enclosure for routing and oddly I've had the reconnects jump up and the send failure rate climb to 50%. The only thing I changed is trying to set the chlorinator output to 0% since the pool was overchlorinated and was trying to let it settle back down to tune it in. Now wondering if I somehow still have some setting off and it's trying to send half the comms over the BAS RS485 and half over the USB dongle? 

It's weird this is still putting up a fight. Something is properly broken though as now it runs for ~5mins then the whole Pi crashes and reboots. I disabled the RS485 Port in the Dashboard>Controller tab and it's stable running (with no comms).

**Reply by tagyoureit** [reply]
@StrikerS7R If you upload a [replay capture](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) we can see if there is anything that sticks out.

**Reply by StrikerS7R** [reply]
> @StrikerS7R If you upload a [replay capture](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) we can see if there is anything that sticks out.

I tried to recreate it and (thankfully) it seems to just be chugging along just fine now. I'll update if it comes back but thank you for the troubleshooting support!
